File: amodeus/models/timetable_element.py
__all__ = ["TimetableElement"]

from datetime import datetime

from .base import BaseIDModel
from .lesson import Lesson
from .location import Location
from .person import Person


class TimetableElement(BaseIDModel):
    lesson: Lesson
    start: datetime
    end: datetime
    location: Location | None
    teachers: list[Person]
    team_name: str | None
    # is_held: bool

File: src/main/kotlin/me/evgfilim1/amodeus/api/models/TimetableElement.kt
package me.evgfilim1.amodeus.api.models

import kotlinx.datetime.Instant
import kotlinx.serialization.Serializable
import me.evgfilim1.amodeus.api.utils.UUID

@Serializable
data class TimetableElement(
    val id: UUID,
    val lesson: Lesson,
    val start: Instant,
    val end: Instant,
    val location: Location?,
    val teachers: List<Person>,
    val team_name: String?,
)
